### PR TITLE
Correct the definitions of precision and recall in 2-4-Logistic.

### PR DESCRIPTION
**Pull request by RyanXinOne**
# Precision and Recall
I think the formula definitions of *Precision* and *Recall* in *2-4-Logistic* are not correct, compared with what tells me on [sklearn Doc](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_fscore_support.html#sklearn.metrics.precision_recall_fscore_support) and [Wikipedia](https://en.wikipedia.org/wiki/Precision_and_recall).

Precision and Recall from Wiki

# Confusion Matrix
There is also a problem that appears in the picture of *Confusion Matrix*. (I did not fix the picture in the commit)

Confusion Matrix

The *Actual Values* should be presented by axis 0 and *Predicted Values* in axis 1. So that calculation results can match the `classification_report` with the correct definition of precision and recall.

Finally, The correct matrix should be:

|||
| :-: | :-: |
| TP | FN |
| FP | TN |

See [Confusion Matrix in sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html?highlight=confusion_matrix#sklearn.metrics.confusion_matrix) and [Wiki](https://en.wikipedia.org/wiki/Confusion_matrix).

### CODE REVIEW

**Comment by jlooper** [comment]
Let me do a bit of research to confirm. I remember going back and forth on this picture representation of the matrix

**Review by jlooper — APPROVED** [review]
Thanks for this edit!

Let’s open a new issue to review that image, I will need to edit it in Procreate.

let’s make sure all translations are updated too

**Comment by RyanXinOne** [comment]
I have opened issue #237 and updated all current translations.

**Comment by jlooper** [comment]
Here's why I struggled (now I remember) - sklearn's axes are different from Wikipedia's. And they say so in the docs, but vaguely. Here's an article: https://towardsdatascience.com/understanding-the-confusion-matrix-from-scikit-learn-c51d88929c79 - it's super....confusing. Can you take a look, see if you think I should redraw the image? There's also a lot of text to rewrite about the axes.

**Comment by RyanXinOne** [comment]
Below is my understanding.

As shown in the article you shared, the default representation of Confusion Matrix in sklearn is
|  |0|1|
|:-:|:-:|:-:|
|0|TN|FP|
|1|FN|TP|

This is true because sklearn treats 1 as true/positive by default.

What [doc](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html?highlight=confusion_matrix#sklearn.metrics.confusion_matrix) says is identical with the matrix above.

> Thus in binary classification, the count of true negatives is C0,0, false negatives is C1,0, true positives is C1,1 and false positives is C0,1.

However, in our lesson, this is not the case. We use 0 to represent color 'orange' which is regarded as "true pumpkins". And, in the graph, what TP refers to is "the model predicts something as a pumpkin and it is actually a pumpkin", which should be represented by (0,0). So this is how it differs.

----------------------------------

The fact can also be proved by the precision(0.83) calculated in `classification_report` above.

The confusion matrix we got is
```python
array([[162,   4],
       [ 33,   0]])
```
precision = tp / (tp + fp) = 162 / (162 + 33) = 0.8307692307692308

----------------------------------

Thus, I still tend to redraw the original image.

**Comment by jlooper** [comment]
Question - should I remove the image? Is it confusing?

**Comment by RyanXinOne** [comment]
The original image did confuse me when I was learning. Because it cannot match the output of the code.

I would recommend to redraw the image, or it should be deleted.

**Review by jlooper — APPROVED** [review]
